source/brailleDisplayDrivers/hims.py

```diff
@@ -418,7 +421,7 @@ def _sendPacket(
 		]
 		packetB = bytearray(b"".join(packet))
 		checksum: int = 0xff & sum(packetB)
-		packetB[-2] = checksum

```

Comment:
Could you add some comments here, it wasn't immediately obvious why this is indexed with `-3'?

It has taken me a (perhaps embarrassingly) long time to spot why this index needs to change. The key to understanding this is recognizing that the last byte string added to the `packet` list (with comment `Packet end`), is repeated (with `*2`) making a string twice as long, held as a single element. It becomes two elements in `packetB`, which is a `bytearray` and has one byte per index. 

A way to demonstrate the difference is with:
```
>>> b = list([b'a', b'0', b'z'*2])
>>> b
[b'a', b'0', b'zz']
>>> b[-2] = b'1'
>>> b
[b'a', b'1', b'zz']
>>> ba = bytearray(b"".join(b))
>>> ba
bytearray(b'a1zz')
>>> ba[-2] = ord('c')
>>> ba
bytearray(b'a1cz')
```

Reply:
Done. Yes, it took me some time to find this out as well.